rag_experiment_accelerator/checkpoint/checkpoint.py

```diff
@@ -0,0 +1,86 @@
+from abc import ABC, abstractmethod
+from typing import Any
+from rag_experiment_accelerator.utils.logging import get_logger
+
+logger = get_logger(__name__)
+
+
+class Checkpoint(ABC):
+    """
+    A Checkpoint is used to cache the results of method calls, enabling the reuse of these results if the same method is called again with the same ID.
+    When a method wrapped by a Checkpoint object is called with an ID that was used before,
```

Comment:
This comment makes it sound like Checkpoint is used like a decorator for a method.
I don't see that being the case.
Have I missed something?

Reply:
I'm not sure it can be a decorator, as it requires the ID to be passed in. This is also one of the reasons we can't use https://joblib.readthedocs.io/en/latest/memory.html - in our case, the arguments do not necessarily identify the call. Hashing the given arguments will not necessarily give us the same value.

Reply:
update: I was able to find a way to pass the ID and now it is used as a decorator

Reply:
this looks much better now, and far easier to use,
well done Yuval